clientv3/kv.go

```diff
@@ -122,21 +122,18 @@ func (kv *kv) Delete(key string) (*DeleteResponse, error) {
 }
 
 func (kv *kv) Compact(rev int64) error {
-	for {
-		r := &pb.CompactionRequest{Revision: rev}
-		_, err := kv.getRemote().Compact(context.TODO(), r)
-		if err == nil {
-			return nil
-		}
-
-		if isRPCError(err) {
-			return err
-		}
+	r := &pb.CompactionRequest{Revision: rev}
+	_, err := kv.getRemote().Compact(context.TODO(), r)
+	if err == nil {
+		return nil
+	}
 
-		if nerr := kv.switchRemote(err); nerr != nil {
-			return nerr
-		}
+	if isRPCError(err) {
+		return err
 	}
+
+	go kv.switchRemote(err)
```

Comment:
Yeah, it seems like the client would manage the connection (probably for all services, not just KV, since the paths are similar) with a goroutine and `client.Close()` could `<-donec` in the usual way.